include/deal.II/base/tensor.h

```diff
@@ -2613,6 +2617,53 @@ cofactor(const Tensor<2, dim, Number> &t)
 }
 
 
+/**
+ * Return the nearest orthogonal matrix using a SVD if the
+ * deteriminant is more than a tolerance away from one.
+ * The orthogonalization is done by combining the products
+ * of the SVD decomposition: $((U*I)*V^T)^T$, where I is the
```

Comment:
Thanks for the review. Yes, you are right. I can (and will) remove the identity matrix. 

What the method does is is basically replacing Sigma with a identity matrix when reassembling it, making the resulting matrix orthogonal. This works because if the input matrix was orthogonal, the decomposition would have been `UIV*`. There is some more info [here](https://en.wikipedia.org/wiki/Singular_value_decomposition#Nearest_orthogonal_matrix). I think this only works well if the matrix is already nearly orthogonal (which is the case for me) and am not sure how well it works if the matrix is far from it, although the question is what it then means to orthogonality. 

Looking at (this page)[https://en.wikipedia.org/wiki/Orthogonalization], I noticed that this is not the recommended way by wikipedia, so I could add a parameter so that users can choose a method to orthogonalize.

My application is that I have a rotation matrix which is changed when computing Latice Preferred Orientation (LPO). After that computation, due to rounding errors, the matrix might not be fully orthogonal anymore and over time these errors accumulate. I use this to keep the rotation matrix orthogonal. I use the SVD method because some other code to compute this, also used this trick to keep the matrix orthogonal.